src/box/box.cc

```diff
@@ -4318,34 +4256,7 @@ box_is_configured(void)
 static void
 box_cfg_xc(void)
 {
-	/* Join the cord interconnect as "tx" endpoint. */
-	fiber_pool_create(&tx_fiber_pool, "tx",
-			  IPROTO_MSG_MAX_MIN * IPROTO_FIBER_POOL_SIZE_FACTOR,
-			  FIBER_POOL_IDLE_TIMEOUT);
-	/* Add an extra endpoint for WAL wake up/rollback messages. */
-	cbus_endpoint_create(&tx_prio_endpoint, "tx_prio", tx_prio_cb, &tx_prio_endpoint);
-
-	rmean_box = rmean_new(iproto_type_strs, IPROTO_TYPE_STAT_MAX);
-	rmean_error = rmean_new(rmean_error_strings, RMEAN_ERROR_LAST);
-
-	gc_init();
-	engine_init();
-	schema_init();
-	replication_init(cfg_geti_default("replication_threads", 1));
-	port_init();
-	iproto_init(cfg_geti("iproto_threads"));
-	sql_init();
-	audit_log_init(cfg_gets("audit_log"), cfg_geti("audit_nonblock"),
-		       cfg_gets("audit_format"), cfg_gets("audit_filter"));
-
-	int64_t wal_max_size = box_check_wal_max_size(cfg_geti64("wal_max_size"));
-	enum wal_mode wal_mode = box_check_wal_mode(cfg_gets("wal_mode"));
-	if (wal_init(wal_mode, cfg_gets("wal_dir"), wal_max_size,
-		     &INSTANCE_UUID, on_wal_garbage_collection,
-		     on_wal_checkpoint_threshold) != 0) {
-		diag_raise();
-	}
-
```

Comment:
Nit: By placing `box_storage_init` before `box_cfg_xc` you could reduce the diff, because this chunk of code wouldn't be moved. It'd be nice if you reshuffled the patch to minimize changes done to the existing code - it'd ease review and preserve the git history.

Reply:
I tried that approach at first but then moved the code intentionally. After I created `box_storage_init()` right above `box_cfg_xc()` (to keep diff small, right), I had hard time trying to split all those init and free calls between `box_init()`, `box_free()`, `box_storage_init()`, and `box_storage_free()`. They were simply too far from each other and I had to jump back and forth in both halves of my screen (it wasn't enough - there were 4 places quite distant from each other, not only 2).

Then I decided better move them to the end of the file together. Firstly, it becomes consistent with other files which have init/free funcs (they are usually at the end). Secondly, it becomes much easier to see in one place what is where created and destroyed. Especially when you want to compare init vs free to ensure the order is reversed in free (at least it should be - here I didn't change the order (inside each function)).